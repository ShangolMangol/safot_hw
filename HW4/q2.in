print "===TEST START===\n";
flatten (Br(0, Br(1, Nil, Nil), Br(2, Nil, Nil)));
map (fn x => 2 * x) (Br(0, Br(1, Nil, Nil), Br(2, Nil, Nil)));