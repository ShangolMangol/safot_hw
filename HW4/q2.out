Standard ML of New Jersey (64-bit) v110.99.2 [built: Sun Mar 20 20:10:23 2022]
[opening hw4_q2.sml]
[opening hw4_q2_def.sml]
datatype 'a tree = Br of 'a * 'a tree * 'a tree | Nil
datatype ('a,'b) union = type1 of 'a | type2 of 'b
val it = () : unit
val flatten = fn : 'a tree -> 'a list
val it = [0,1,2] : int list
val it = [0,1,3,2] : int list
hw4_q2.sml:21.8-25.90 Warning: match nonexhaustive
          (f,notDone (Br (node,left,right)),queue,finished) => ...
          (f,doneLeft (Br (node,left,right)),queue,finished) => ...
          (f,doneBoth (Br (node,left,right)),nil,a1 :: a2 :: nil) => ...
          (f,doneBoth (Br (node,left,right)),x :: queue,a1 :: a2 :: finished) => ...
          (f,notDone Nil,x :: queue,finished) => ...
  
val map = fn : ('a -> 'b) -> 'a tree -> 'b tree
- ===TEST START===
val it = () : unit
val it = [0,1,2] : int list
val it = Br (0,Br (2,Nil,Nil),Br (4,Nil,Nil)) : int tree

