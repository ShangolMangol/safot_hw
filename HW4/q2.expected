val it = () : unit
val it = [0,1,2] : int list
val it = Br (0,Br (2,Nil,Nil),Br (4,Nil,Nil)) : int tree

